genie-core/src/main/java/com/netflix/genie/core/jobs/JobKillReasonFile.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.core.jobs;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+/**
+ * Class that represents the structure of the kill-reason file created when a job is killed.
+ *
+ * @author mprimi
+ * @since 3.0.8
+ */
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
Use `@Data` should suffice for all these.

I prefer Dto's to be read only. Perhaps what you should do here (if it's intended to be bound to json) is just create the constructor yourself and annotate it with `@JsonCreator` that way jackson can bind it but we can still make it read only.